misc.h

```diff
@@ -37,19 +37,114 @@
+
+	#define EEPROM_MR_FREQ_OFF       0x2000  // Start of the channel relocation
+	#define EEPROM_MR_FREQ_LEN       (MR_CHANNELS_COUNT * 16)
+
+	#define EEPROM_MR_ATTR_OFF       (EEPROM_MR_FREQ_OFF + EEPROM_MR_FREQ_LEN)
+	#define EEPROM_MR_ATTR_LEN       (MR_CHANNELS_COUNT)
+
+	#define EEPROM_VFO_ATTR_OFF      (EEPROM_MR_ATTR_OFF + EEPROM_MR_ATTR_LEN)
+	#define EEPROM_VFO_ATTR_LEN      (VFO_CHANNELS_COUNT)
+
+	#define EEPROM_MR_SETTINGS_OFF   (EEPROM_VFO_ATTR_OFF + EEPROM_VFO_ATTR_LEN) // Screen CH A, MR CH A, Freq CH A, Screen CH B, MR CH B, Freq CH B, NOAA CH A, NOAA CH B
+	#define EEPROM_MR_SETTINGS_LEN   0x0010
+
+	#define EEPROM_MR_NAME_OFF       (EEPROM_MR_SETTINGS_OFF + EEPROM_MR_SETTINGS_LEN)
+	#define EEPROM_MR_NAME_LEN       (EEPROM_MR_FREQ_LEN)
+	#define EEPROM_LAST              (EEPROM_MR_NAME_OFF + EEPROM_MR_NAME_LEN)
+	static const uint16_t MR_CHANNEL_FIRST   = 0;
+	static const uint16_t MR_CHANNEL_LAST    = (MR_CHANNELS_COUNT - 1);
+	static const uint16_t VFO_CHANNEL_FIRST = MR_CHANNELS_COUNT;
+	static const uint16_t VFO_CHANNEL_LAST  = (MR_CHANNELS_COUNT + VFO_CHANNELS_COUNT - 1);
+	static const uint16_t NOAA_CHANNEL_FIRST = (MR_CHANNELS_COUNT + VFO_CHANNELS_COUNT);
+	static const uint16_t NOAA_CHANNEL_LAST  = (MR_CHANNELS_COUNT + VFO_CHANNELS_COUNT + FM_CHANNELS_COUNT -1);
+	static const uint16_t LAST_CHANNEL       = (MR_CHANNELS_COUNT + VFO_CHANNELS_COUNT + FM_CHANNELS_COUNT);
+#else
+	typedef uint8_t channel_t;
+
+	#define EEPROM_SIZE               0x2000
+	#define VFO_CHANNELS_COUNT        7
+	#define FM_CHANNELS_COUNT         20
+	#define MR_CHANNELS_COUNT         200
+	#define MAX_CHANNEL               0xff
+	#define ALL_CHANNELS_COUNT        (VFO_CHANNELS_COUNT + FM_CHANNELS_COUNT + MR_CHANNELS_COUNT)
+
+	#define EEPROM_MR_FREQ_OFF        0x0000
+	#define EEPROM_MR_FREQ_LEN        0x0C80
+
+	#define EEPROM_MR_ATTR_OFF        0x0D60
+	#define EEPROM_MR_ATTR_LEN        0x00C8
+
+	#define EEPROM_VFO_ATTR_OFF       0x0E28
+	#define EEPROM_VFO_ATTR_LEN       0x0007
+
+	#define EEPROM_MR_SETTINGS_OFF    0x0E80
+	#define EEPROM_MR_SETTINGS_LEN    0x0008
+
+	#define EEPROM_MR_NAME_OFF        0x0F50
+	#define EEPROM_MR_NAME_LEN        0x0C80
+#endif  // ENABLE_ALT_CHANNELS
+
+#define EEPROM_VFO_FREQ_OFF       0x0C80
+#define EEPROM_VFO_FREQ_LEN       0x00E0
+
+#define EEPROM_FM_CHANNELS_OFF    0x0E40
+#define EEPROM_FM_CHANNELS_LEN    0x0028
+
+#define EEPROM_OPTIONS_0E70_OFF   0x0E70
+#define EEPROM_OPTIONS_0E70_LEN   0x0008
+
+#define EEPROM_OPTIONS_0E78_OFF   0x0E78
+#define EEPROM_OPTIONS_0E78_LEN   0x0008
+
+#define EEPROM_FM_CH_SETTINGS_OFF 0x0E88
+#define EEPROM_FM_CH_SETTINGS_LEN 0x0003
+
+#define EEPROM_WELCOME_STR1_OFF   0x0EB0
+#define EEPROM_WELCOME_STR1_LEN   0x0010
+#define EEPROM_WELCOME_STR2_OFF   0x0EC0
+#define EEPROM_WELCOME_STR2_LEN   0x0010
+
+#define EEPROM_CODE_OPTIONS_OFF   0x0ED0
+#define EEPROM_CODE_OPTIONS_LEN   0x000B
```

Comment:
Lengths in this section are not needed, they are not use anywhere

Reply:
My intentions is to have hints for the expected size, even not used in the code...